index.js

```diff
@@ -192,29 +191,57 @@ function getMetrics({ projectKey, startTime, endTime, prNum }) {
  * - job annotation
  * @method getProjectData
  * @param  {Object}     config
- * @param  {Object}     [config.annotations]        Screwdriver job annotations
+ * @param  {Object}     [config.scope]              Coverage scope (pipeline or job)
  * @param  {Boolean}    config.enterpriseEnabled    If enterprise is enabled
  * @param  {String}     config.jobId                Screwdriver job ID
  * @param  {String}     config.jobName              Screwdriver job name
  * @param  {String}     config.pipelineId           Screwdriver pipeline ID
  * @param  {String}     config.pipelineName         Screwdriver pipeline name
+ * @param  {String}     [config.projectKey]         Sonar project key
+ * @param  {String}     [config.prNum]              PR number
+ * @param  {String}     [config.prParentJobId]      Screwdriver PR parent job ID
  * @return {Object}                                 Sonar project key, project name, and username
  */
-function getProjectData({ annotations, enterpriseEnabled, jobId, jobName, pipelineId,
-    pipelineName }) {
-    // Figure out default scope: pipeline scope for enterprise edition, job scope for everything else
-    const defaultScope = enterpriseEnabled ? 'pipeline' : 'job';
-    // Use user-configured scope or default scope
-    const scope = (annotations && annotations[COVERAGE_SCOPE_ANNOTATION]) || defaultScope;
-
-    if (scope === 'pipeline') {
+function getProjectData({ scope, enterpriseEnabled, jobId: buildJobId,
+    jobName: buildJobName, pipelineId, pipelineName, projectKey, prParentJobId, prNum }) {
+    let jobId = buildJobId;
+    let jobName = buildJobName;
+
+    // Determine scope based on projectKey
+    if (projectKey) {
+        const [projectScope, id] = projectKey.split(':');
+        const projectName = projectScope === 'pipeline' ?
+            pipelineName : `${pipelineName}:${jobName}`;
+        const username = (prNum && projectScope === 'job') ?
```

Comment:
I think we we need a separate user for pr jobs only for non-enterprise mode ?  In enterprise `job` scope it can take the same user as the parent Job

Reply:
updated